src/main/java/org/folio/innreach/batch/contribution/InitialContributionJobConsumerContainer.java

```diff
@@ -0,0 +1,114 @@
+package org.folio.innreach.batch.contribution;
+
+import lombok.RequiredArgsConstructor;
+import lombok.Setter;
+import lombok.extern.log4j.Log4j2;
+import org.apache.kafka.clients.consumer.ConsumerRecord;
+import org.apache.kafka.common.serialization.Deserializer;
+import org.folio.innreach.batch.contribution.listener.ContributionExceptionListener;
+import org.folio.innreach.domain.dto.folio.inventorystorage.InstanceIterationEvent;
+import org.folio.innreach.external.exception.ServiceSuspendedException;
+import org.springframework.kafka.core.ConsumerFactory;
+import org.springframework.kafka.core.DefaultKafkaConsumerFactory;
+import org.springframework.kafka.listener.ConcurrentMessageListenerContainer;
+import org.springframework.kafka.listener.ContainerProperties;
+import org.springframework.kafka.listener.DefaultErrorHandler;
+import org.springframework.util.backoff.BackOff;
+import org.springframework.util.backoff.FixedBackOff;
+
+import java.net.SocketTimeoutException;
+import java.util.HashMap;
+import java.util.Map;
+
+import static org.apache.kafka.clients.consumer.ConsumerConfig.ENABLE_AUTO_COMMIT_CONFIG;
+
+@Log4j2
+@Setter
+@RequiredArgsConstructor
+public class InitialContributionJobConsumerContainer {
+
+  private static Map<String, ConcurrentMessageListenerContainer<String, InstanceIterationEvent>> consumersMap =
+    new HashMap<>();
+
+
+  private final Map<String,Object> consumerProperties;
+  private final String topic;
+  private final Deserializer<String> keyDeserializer;
+  private final Deserializer<InstanceIterationEvent> valueDeserializer;
+
+  private final Long interval;
+
+  private final Long maxAttempts;
+
+  private final int THREAD_SIZE = 2;
```

Comment:
_in general code could be prettier to increase its readability and maintainability_. E.g.: even within current example there are two omissions: 1) for such format it should be static final field; 2) for increasing meaning - it could be something like: "concurrent consumers amount" or "concurrent container size" as once noticed THREAD_SIZE constant devs who will maintain the codebase will likely imply Thread Pool, but it's just a property for concurrency of ConcurrentMessageListenerContainer.